src/dex/woo-fi/woo-fi.ts

```diff
@@ -0,0 +1,556 @@
+                ),
+              );
+            } else {
+              // One of them must be quoteToken
+              return null;
+            }
+          }
+        }
+
+        const unit = _prices[0];
+
+        result.push({
+          unit,
+          prices: [0n, ..._prices.slice(1)],
+          data: {},
+          poolIdentifier: allowedPairIdentifier,
+          exchange: this.dexKey,
+          gasCost: WOO_FI_GAS_COST,
+          poolAddresses: [this.exchangeAddress],
+        });
+      }
+      return result;
+    } catch (e) {
+      this.logger.error(
+        `Error_getPricesVolume ${srcToken.symbol || srcToken.address}, ${
+          destToken.symbol || destToken.address
+        }, ${side}:`,
+        e,
+      );
+      return null;
+    }
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: WooFiData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    if (side === SwapSide.BUY) throw new Error(`Buy not supported`);
+
+    return {
+      targetExchange: this.exchangeAddress,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: WooFiData,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    if (side === SwapSide.BUY) throw new Error(`Buy not supported`);
+
+    const _srcToken = srcToken.toLowerCase();
+    const _destToken = destToken.toLowerCase();
+
+    let funcName: string;
+    let _amount: string;
+    let baseToken: string;
+    if (_srcToken === this.quoteTokenAddress) {
+      baseToken = _destToken;
+      funcName = 'sellQuote';
+      _amount = srcAmount;
+    } else if (_destToken === this.quoteTokenAddress) {
+      baseToken = _srcToken;
+      funcName = 'sellBase';
+      _amount = srcAmount;
+    } else {
+      throw new Error(
+        `srcToken ${srcToken} or destToken ${destToken} must be quoteToken`,
+      );
+    }
+
+    const swapData = this.wooIfaces.PP.encodeFunctionData(funcName, [
+      baseToken, // baseToken
+      _amount, // amount
+      MIN_CONVERSION_RATE, // minAmount
+      this.augustusAddress, // to
+      NULL_ADDRESS, // rebateTo
+    ]);
+
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      this.exchangeAddress,
+    );
+  }
+
+  async updatePoolState(): Promise<void> {
+    await this.getState();
+  }
+
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    const wrappedTokenAddress = wrapETH(
+      { address: tokenAddress, decimals: 0 },
+      this.network,
+    ).address.toLowerCase();
+
+    const filteredToken = Object.values(this.tokenByAddress!).filter(
+      token => token.address === wrappedTokenAddress,
+    );
+
+    if (filteredToken.length === 0) return [];
```

Comment:
This doesn't make sense to me, instead, it should be
```suggestion
  if(!this.tokenByAddress[wrappedTokenAddress])
     return [];
```

Reply:
Fixed. I am not sure, why I did that way :)